Tools/CISupport/ews-build/steps.py

```diff
@@ -3940,15 +3957,20 @@ def evaluateCommand(self, cmd):
                                                 UploadTestResults(),
                                                 ExtractTestResults()])
             self.finished(WARNINGS)
+
         else:
             self.build.addStepsAfterCurrentStep([
                 ArchiveTestResults(),
                 UploadTestResults(),
                 ExtractTestResults(),
                 ValidateChange(verifyBugClosed=False, addURLs=False),
-                KillOldProcesses(),
-                ReRunWebKitTests(),
             ])
+            if GitHub.NO_FAILURE_LIMITS not in self.getProperty('labels', []):
+                self.build.addStepsAfterCurrentStep([
```

Comment:
This isn't going to work in the way we expect it to, because of this buildbot bug: https://github.com/buildbot/buildbot/issues/4874 

Also, the uat instance testing url http://ews-build.webkit-uat.org/#/builders/4/builds/21919 doesn't test this part. Would be good to test this logic of uat instance.

The way we typically tackle this is to maintain a variable like steps_to_add and keep adding steps to that list and finally do a single call to self.build.addStepsAfterCurrentStep with that variable.